Duplicati/Library/Main/Operation/Common/BackendHandler.cs

```diff
@@ -435,7 +434,7 @@ public Task<Library.Utility.TempFile> GetFileForTestingAsync(string remotename,
         private async Task<bool> DoPut(FileEntryItem item, bool updatedHash = false)
         {
             // If this is not already encrypted, do it now
-            await item.Encrypt(m_options).ConfigureAwait(false);
+            item.Encrypt(m_options);

```

Comment:
The [other invocation](https://github.com/duplicati/duplicati/pull/3309/files#diff-59e4ce9a1a5775dcf7c9d1fa42821830R224) of `Encrypt` is performed on the thread pool using `Task.Run`.  I'm not sure if it makes sense to do so here as well?